src/main/java/com/google/codeu/servlets/StatsPageServlet.java

```diff
@@ -23,22 +23,16 @@ public void init() {
     datastore = new Datastore();
   }
 
-  /**
-   * Responds with site statistics in JSON. 
-   * Example JSON object: {"messageCount": 5, "userCount": 3}
-   */
   @Override
-  public void doGet(HttpServletRequest request, HttpServletResponse response)
-      throws IOException {
-
-    response.setContentType("application/json");
+  public void doGet(HttpServletRequest request, HttpServletResponse response) throws IOException, ServletException {
 
     int messageCount = datastore.getTotalMessageCount();
     int userCount = datastore.getUsers().size();
 
-    JsonObject jsonObject = new JsonObject();
-    jsonObject.addProperty("messageCount", messageCount);
-    jsonObject.addProperty("userCount", userCount);
-    response.getOutputStream().println(jsonObject.toString());
+    request.setAttribute("messageCount", messageCount);
+    request.setAttribute("userCount", userCount);
+    
+    request.getRequestDispatcher("/WEB-INF/views/stats.jsp").forward(request, response);
```

Comment:
Could you comment on what this does?

Reply:
This code takes the logic that used to load the stats in the client via JavaScript, and renders it server-side instead. If you look at the stats.jsp page, it renders the attributes set on lines 32 and 33.